### DISCUSSION TITLE
How to detect open STP orders?

### DISCUSSION THREAD

**Discussion by ferraro** [erdewit/ib_insync#691] [Q&A]
I'm writing a position manager python script, that checks if all open future positions have stop losses set.
And if not, it should close the position. However, I'm unable to detect if a STP order is active or not?
Even old STP orders, which have been executed have the status "Filled". And non executed STP order have too the status "Filled". I simply cannot find a way to find out whether an STP order is still active or has already been triggered.

**Comment by gsy1xy** [comment]
Instead, I check the position during each round to see what happened, and cancel the old stp order, then place new stop order if needed.

**Comment by ferraro** [comment]
I found a solution which works:
def has_stop_loss_order(ib, position):
    """
    Prüft, ob für eine gegebene Position eine aktive Stop-Loss-Order existiert.
    """
    trades = ib.reqAllOpenOrders()
    for trade in trades:
        # Prüfe für jede aktive Order, zu welcher Position sie gehört, und ob es sich um eine STP-Order handelt.
        if (trade.contract.localSymbol == position.contract.localSymbol and
            trade.order.orderType == 'STP' and
            trade.orderStatus.status == 'PreSubmitted'):
            print_with_timestamp(f'Stop-Loss-Order für Position {position.contract.localSymbol} gefunden.')
            return True
    return False

**Comment by ferraro** [comment]
Problem solved with the command ib.reqAllOpenOrders(), which gives back open STP orders with the related contract and its local symbol name.
